Improve `LockOnNonEnclosingClassLiteral` documentation (#61)
PicnicSupermarket/error-prone: docs/bugpattern/LockOnNonEnclosingClassLiteral.md

Having a lock on a class, other than the enclosing class of the code block, can
unintentionally prevent the locked class from being used properly when other

> classes effectively lock on the same resource. This can be time-consuming to
> handle from a maintainability perspective to ensure the `synchronized` blocks
> are working as expected. Hence, locking on a class other than the enclosing
> class of the `synchronized` code block is discouraged by Error Prone. Locking
> on the enclosing class or an instance is a preferred practice.

Reviewer: Sorry, saw one more grammatical opportunity to be a bit more clear (preventing 'to handle [...] to ensure').

```suggestion
classes effectively lock on the same resource. From a maintainability
perspective, it can be time-consuming to ensure the `synchronized` blocks are
working as expected. Hence, locking on a class other than the enclosing class
of the `synchronized` code block is discouraged by Error Prone. Locking on the
enclosing class or an instance is a preferred practice.
```

Also, one of the reasons why this is discourages, can also be the opaqueness of such issues and it being hard to identify, which is not really captured well. At least, that is what 'time consuming and exhausting' made me think about. But I couldn't come up with the right words for that. Left it out for now, but if you feel different, feel free to tweak.

Author: Thanks for the suggestions! 😄

Lock on the class other than the enclosing class of the code block can
unintentionally prevent the locked class or other classing using the same lock
being used properly. An issue can be caused when the locked class is being edited
or deleted. And it can also be exhausting and time-consuming for the others who
are using the same class literal as lock, to make sure the `synchronized` blocks
can work properly as expected. Hence, locking on the class other than the
enclosing class of the `synchronized` code block is discouraged by Error Prone.
Locking on the enclosing class or an instance is a preferred practice.

For example, a lock on `Other.class` rather than `Example.class` will trigger an
Error Prone warning:

> ```

Reviewer 2: ```suggestion
```java
```
And likewise below :)

class Example {

>   method() {

Reviewer 2: ```suggestion
  void method() {
```
And likewise below :)

Having a lock on a class, other than the enclosing class of the code block, can
unintentionally prevent the locked class from being used properly. This also
holds for other classes that are using the same lock.

> An issue can arise when the locked class is being edited or deleted.

Reviewer: Not sure if this sentence holds any value compared to the sentences above TBH.

Author: Yeah makes sense, I think I tried to stay to close to the original content.

> Having a lock on a class, other than the enclosing class of the code block, can
> unintentionally prevent the locked class from being used properly. This also
> holds for other classes that are using the same lock.
> An issue can arise when the locked class is being edited or deleted.
> Additionally, when others are using the same class literal as lock, it can be
> exhausting and time-consuming for others to ensure that the `synchronized`
> blocks are working as expected. Hence, locking on a class other than the
> enclosing class of the `synchronized` code block is discouraged by Error Prone.
> Locking on the enclosing class or an instance is a preferred practice.

Reviewer: I was having a bit too much trouble with these paragraph. What is really the issue was not clear to me. Furthermore, some sentences are a bit obsolete/saying the same thing. I think we can minimize this to:

```suggestion
Having a lock on a class, other than the enclosing class of the code block, can 
unintentionally prevent the locked class from being used properly when other
classes effectively lock on the same resource. This can be time-consuming to
handle from a maintainability perspective to ensure the `synchronized` blocks
are working as expected. Hence, locking on a class other than the enclosing
class of the `synchronized` code block is discouraged by Error Prone. Locking
on the enclosing class or an instance is a preferred practice.
```

Author: Nice one, let's do it.
